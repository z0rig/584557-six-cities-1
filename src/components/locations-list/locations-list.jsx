import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import {actionCreators} from "../../reducer/current-city/current-city";
import {getLocations, getCurrentCity} from "../../reducer/offers/selectors";

const LocationsList = (props) => {
  const {locations, currentCity, changeCurrentCity} = props;

  return (
    <div className="cities tabs">
      <section className="locations container">
        <ul className="locations__list tabs__list">
          {[...locations.keys()].map((it, ind) => {
            return (
              <li className="locations__item" key={ind}>
                <a
                  className={`locations__item-link tabs__item${
                    it === currentCity ? ` tabs__item--active` : ``
                  }`}
                  onClick={() => {
                    changeCurrentCity(it);
                  }}
                >
                  <span>{it}</span>
                </a>
              </li>
            );
          })}
        </ul>
      </section>
    </div>
  );
};

LocationsList.propTypes = {
  currentCity: PropTypes.string,
  changeCurrentCity: PropTypes.func,
  locations: PropTypes.object
};

const mapStateToProps = (state) => {
  return {
    currentCity: getCurrentCity(state),
    locations: getLocations(state)
  };
};

const mapDispatchToProps = (dispatch) => ({
  changeCurrentCity: (city) => dispatch(actionCreators.changeCurrentCity(city))
});

export {LocationsList};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(LocationsList);
