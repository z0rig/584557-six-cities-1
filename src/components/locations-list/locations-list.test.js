import React from "react";
import renderer from "react-test-renderer";
import {LocationsList} from "./locations-list.jsx";

it(`LocationsList correctly renders`, () => {
  const tree = renderer
    .create(<LocationsList currentCity={`Amsterdam`} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
