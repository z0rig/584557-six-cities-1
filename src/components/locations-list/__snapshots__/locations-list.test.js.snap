// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationsList correctly renders 1`] = `
<div
  className="cities tabs"
>
  <section
    className="locations container"
  >
    <ul
      className="locations__list tabs__list"
    >
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item tabs__item--active"
          onClick={[Function]}
        >
          <span>
            Amsterdam
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item"
          onClick={[Function]}
        >
          <span>
            Tbilisi
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item"
          onClick={[Function]}
        >
          <span>
            Saint Petersburg
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item"
          onClick={[Function]}
        >
          <span>
            Beijing
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item"
          onClick={[Function]}
        >
          <span>
            Buenos Aires
          </span>
        </a>
      </li>
      <li
        className="locations__item"
      >
        <a
          className="locations__item-link tabs__item"
          onClick={[Function]}
        >
          <span>
            Summerside
          </span>
        </a>
      </li>
    </ul>
  </section>
</div>
`;
